
 This patch automatically adds the Bacula database and user name to
 the default make_catalog_backup Run script call line in the bacula-dir.conf
 file.

 Apply it to version 2.2.8 or earlier with:

 cd <bacula-source>
 patch -p0 <2.2.8-bacula-conf.patch
 ./configure <your-options>
 make
 ...
 make install


Index: src/dird/bacula-dir.conf.in
===================================================================
--- src/dird/bacula-dir.conf.in	(revision 6327)
+++ src/dird/bacula-dir.conf.in	(working copy)
@@ -61,7 +61,11 @@
   FileSet="Catalog"
   Schedule = "WeeklyCycleAfterBackup"
   # This creates an ASCII copy of the catalog
-  RunBeforeJob = "@scriptdir@/make_catalog_backup bacula bacula"
+  # WARNING!!! Passing the password via the command line is insecure.
+  # see comments in make_catalog_backup for details.
+  # Arguments to make_catalog_backup are:
+  #  make_catalog_backup <database-name> <user-name> <password> <host>
+  RunBeforeJob = "@scriptdir@/make_catalog_backup @db_name@ @db_user@"
   # This deletes the copy of the catalog
   RunAfterJob  = "@scriptdir@/delete_catalog_backup"
   Write Bootstrap = "@working_dir@/BackupCatalog.bsr"
